ios/FluentUI.Demo/FluentUI.Demo/Demos/ColorDemoController.swift

```diff
@@ -140,26 +139,31 @@ class ColorDemoController: UIViewController {
     }
 
     private lazy var segmentedControl: SegmentedControl = {
-        let segmentedControl = SegmentedControl(items: colorProviderThemedWindowTypes.map({ return SegmentItem(title: $0.name) }), style: .primaryPill)
+        let segmentedControl = SegmentedControl(items: colorProviderThemes.map({ return SegmentItem(title: $0.name) }), style: .primaryPill)
         segmentedControl.addTarget(self, action: #selector(segmentedControlValueChanged(sender:)), for: .valueChanged)
         return segmentedControl
     }()
 
     @objc private func segmentedControlValueChanged(sender: Any) {
         if let segmentedControl = sender as? SegmentedControl {
-            let windowType = colorProviderThemedWindowTypes[segmentedControl.selectedSegmentIndex].windowType
-            let colorThemeHost = view.window?.windowScene?.delegate as? ColorThemeHosting
-
-            if let navigationController = navigationController {
-                navigationController.popViewController(animated: false)
-                colorThemeHost?.updateToWindowWith(type: windowType, pushing: self)
+            let window = colorProviderThemes[segmentedControl.selectedSegmentIndex].demoColorTheme.window
+            if let colorProvider = window as? ColorProviding {
+                Colors.setProvider(provider: colorProvider, for: window)
             }
         }
+
+        tableView.reloadData()
     }
 
-    private let colorProviderThemedWindowTypes: [(name: String, windowType: UIWindow.Type)] = [("Default", DemoColorThemeDefaultWindow.self),
-                                                                                               ("Green", DemoColorThemeGreenWindow.self),
-                                                                                               ("None", UIWindow.self)]
+    private let tableView = UITableView(frame: .zero, style: .grouped)
+        private let colorProviderThemes: [(name: String, demoColorTheme: DemoColorTheme)] = [("Default", DemoColorTheme.init(window: DemoColorThemeDefaultWindow(), windowType: DemoColorThemeDefaultWindow.self)),
+                                                                                             ("Green", DemoColorTheme.init(window: DemoColorThemeGreenWindow(), windowType: DemoColorThemeGreenWindow.self)),
+                                                                                             ("None", DemoColorTheme.init(window: UIWindow(), windowType: UIWindow.self))]
+
+    struct DemoColorTheme {
```

Comment:
this new struct should be defined somewhere else outside of this {} bracket? also mark private?